### PR TITLE
Config changes, generation for Windows Installer

### PR DESCRIPTION
**Pull request by tobylane**
**Describe what the proposed change does**
- Clarify the comment about the TH folder
- Add direct zoom and campaign header to windows template. Was the campaign part deliberately missing?
- Remove the paragraph about folders as they're above.

### CODE REVIEW

File: WindowsInstaller/config_template.txt
Comment (Alberth289346) [inline WindowsInstaller/config_template.txt:135]: As we state this as available, we're a distributor. There is a proper license attached to this data?

Also, if we distribute it, why not put it at GH itself, this url doesn't give much confidence imho.

File: WindowsInstaller/config_template.txt
Comment (Alberth289346) [inline WindowsInstaller/config_template.txt:199]: Euhm, below ?

File: WindowsInstaller/config_template.txt
Comment (Alberth289346) [inline WindowsInstaller/config_template.txt:202]: This doesn't look related to the campaign menu and thus should be clearly separated from it.

File: WindowsInstaller/config_template.txt
Comment (Alberth289346) [inline WindowsInstaller/config_template.txt:191]: I think it's default now? Not so experimental thus :)

File: WindowsInstaller/config_template.txt
Comment (tobylane) [reply WindowsInstaller/config_template.txt:135]: I asked the creator on his [youtube](https://www.youtube.com/user/zenithalravage) channel, on a video now private. He gave this link. We could ask for permission to host it ourselves?

**Comment by tobylane** [comment]
Player name needs to be left out, it'll be set to the computer username by the time the user sees it. Agree with your other points.

I copied the output config over to the Windows template. Is there a reason they were different? Column width now 100.

New commit removes trailing whitespaces so that the output is acceptable in a commit, ie when you add a new option in config_finder.lua, run it to get the clean config.txt and copy that to WindowsInstaller/config_template.lua.

File: WindowsInstaller/config_template.txt
Comment (TheCycoONE) [reply WindowsInstaller/config_template.txt:253]: At least in the US and Germany we would not be infringing by linking to copyrighted material. That was the aspect of SOPA that had the tech companies so up in arms all those years ago.

Comment (TheCycoONE) [reply WindowsInstaller/config_template.txt:253]: That said I'm leaning against calling it out there. The wiki or the discussion group seems like more appropriate forums.

**Comment by TheCycoONE** [comment]
If you can figure out a way to avoid the redundant config_template.txt in the windows installer, I'd really like to drop it.

**Comment by tobylane** [comment]
Removed the music link and added the small lua program.

**Comment by TheCycoONE** [comment]
Nice! Thanks.

File: WindowsInstaller/config_template.txt
Comment (TheCycoONE) [inline WindowsInstaller/config_template.txt:334]: I'd like to think I'm at least moderately familiar with CorsixTH and it's various config settings / menus, but the instructions in this CAMPAIGN MENU section don't make any sense to me. I suspect for a person less familiar they might even be actively harmful.

The name itself has to go in a player_name property which we never call out. We don't tell the user that by campaign menu we mean a new game, not the custom campaign selection. The idea was probably that player_name would be injected below, but that doesn't happen either in the template, or if you start the game and quit without setting the name.

suggest tweaking `CorsixTH/Lua/dialogs/resizables/new_game.lua:L102 to ignore blank names, add `player_name = ''` to the default config and put `player_name=[[' .. + config_values.player_name .. ]]` here.

Not keen on the title either, we don't have a campaign menu.

**Review by TheCycoONE — APPROVED** [review]
campaign menu stuff predates your change, so I'll merge as is unless you feel like you want to clean it up.

**Comment by tobylane** [comment]
Done that. It will only use the first 12 characters of a name because after that it overlaps with the money in the yearly reports. It will still overlap if someone has a million and a max length name, but I think that's rare. If the report UI ever gets bigger then consider upping this limit.

File: CorsixTH/Lua/config_finder.lua
Comment (lewri) [inline CorsixTH/Lua/config_finder.lua:524]: This is still useful. But perhaps in the wrong place.

File: WindowsInstaller/config_template.txt
Comment (lewri) [inline WindowsInstaller/config_template.txt:10]: Hello random spaces :D

File: WindowsInstaller/config_template.txt
Comment (lewri) [inline WindowsInstaller/config_template.txt:49]: This should fit on one line? Unless you're emphasising "SDL_mixer library"

File: WindowsInstaller/config_template.txt
Comment (lewri) [inline WindowsInstaller/config_template.txt:123]: Is this for the heat map?

File: WindowsInstaller/config_template.txt
Comment (lewri) [inline WindowsInstaller/config_template.txt:144]:
```suggestion
-- Intro movie: Enabled by default
```

File: WindowsInstaller/config_template.txt
Comment (lewri) [inline WindowsInstaller/config_template.txt:203]: Most users are Windows for this config no? Probably best to use the Windows format.

``C:\Program Files (x86)\Theme Hospital``
^ other variants are available, annoyingly. And assuming it let you install the game on Windows. Otherwise use the GOG default I guess of ``C:\Program Files (x86)\GOG.com\Theme Hospital``

File: WindowsInstaller/config_template.txt
Comment (lewri) [inline WindowsInstaller/config_template.txt:210]: Again if this is for Windows, fonts are located at ``C:\Windows\Fonts``

**Review by lewri — COMMENTED** [review]
Where you've snipped col width to 80 I think you no longer have your menu headers centrally aligned...

Comments generally apply across config_finder and config_template where both would apply.

File: CorsixTH/Lua/config_finder.lua
Comment (tobylane) [reply CorsixTH/Lua/config_finder.lua:516]: There's no such game menu, the closest one is New Game.

File: CorsixTH/Lua/config_finder.lua
Comment (tobylane) [reply CorsixTH/Lua/config_finder.lua:517]: I'll come back to that, as at first glance the character limit doesn't work there.

File: WindowsInstaller/config_template.txt
Comment (tobylane) [reply WindowsInstaller/config_template.txt:210]: Currently, the font loading will only try the OS specific fallback paths if this is unset. I'll fix it later.

**Review by lewri — APPROVED** [review]

File: CorsixTH/Lua/config_finder.lua
Comment (TheCycoONE) [reply CorsixTH/Lua/config_finder.lua]: I doubt it, the tracks are likely dual copyright to ZR and EA which would make them illegal for anyone to distribute.

File: WindowsInstaller/config_template.txt
Comment (TheCycoONE) [reply WindowsInstaller/config_template.txt:122]: Yes

File: WindowsInstaller/config_template.txt
Comment (TheCycoONE) [reply WindowsInstaller/config_template.txt]: This file is automatically generated from config_finder.lua ... actually it is curious that toby's name ended up in it.

File: WindowsInstaller/config_template.txt
Comment (TheCycoONE) [inline WindowsInstaller/config_template.txt:10]: Dash lengths and line wrapping should match

File: WindowsInstaller/config_template.txt
Comment (lewri) [reply WindowsInstaller/config_template.txt:122]: Perhaps this needs a bit more clarification then. Originally I thought it was referring to colour-blindness or overall screen colour (think f.lux)

File: CorsixTH/Lua/config_finder.lua
Comment (TheCycoONE) [inline CorsixTH/Lua/config_finder.lua:517]: This is going to wipe out player_name when it's saved again right? needs to be player_name = [[ .. config_values.player_name .. ']]\n'

**Comment by tobylane** [comment]
Took out the player name length limit, there's more research needed for that.  
config_template generation now uses the defaults.

**Comment by lewri** [comment]
This one all good now Toby?

**Comment by tobylane** [comment]
There is the need to update this section of the installer, but I'd rather leave that to or collaborate with someone on Windows. Folder and screen resolution would be basic replaces, language could be /[[English]]/"[[" + $(install_language) + "]]"/ if concatenation is allowed, and not sure on fullscreen. Worth holding this pr if there's no quick solution before the hotfix release. https://github.com/CorsixTH/CorsixTH/blob/master/WindowsInstaller/Win32Script.nsi#L250

**Comment by tobylane** [comment]
Reduced what the generate script does for no benefit. Ready.

**Comment by TheCycoONE** [comment]
Will merge after I test the windows installer with this.

**Comment by lewri** [comment]
@tobylane I believe #2050 might've been merged with a mismatching format to this PR

**Comment by tobylane** [comment]
Rebased, a minor reformat to the changes of that PR.

**Pull request merged**
